app/splits/[id]/split-game-match-details-panel.tsx

```diff
@@ -9,45 +23,65 @@ function SplitGameMatchDetailsPanel({
   splitGame: SplitGame;
   onUpdate: (splitGame: SplitGame) => void;
 }) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
   function onWinChanged(win: boolean | null) {
     onUpdate({
       ...splitGame,
       win: win,
     });
   }
 
-  function onOpponentClick() {
-    console.log("Opponent clicked");
+  function onOppositeChampionSelected(championData: ChampionData) {
+    onUpdate({
+      ...splitGame,
+      opponentChampionId: championData.id,
+    });
+
+    onClose();
```

Comment:
I suggest making a distinction from the event and the handler. Might be confusing for some to use a function called `onClose` if it's not an event watcher/invoker
```suggestion
  function handleOppositeChampionSelected(championData: ChampionData) {
    handleUpdate({
      ...splitGame,
      opponentChampionId: championData.id,
    });

    handleClose(); // or close();
```